### PR TITLE
Tabbed encounter supports RTL languages

### PR DESCRIPTION
**Pull request by matrix-amiel**
Hi @sunsetsystems , @bradymiller 

About issue https://github.com/openemr/openemr/issues/1272

I suggests here solution for RTL problem in the new tabs of encounter screen.
I also added minor css improvement that minimizes a padding of wrapper frame and make a UI nicer.

Amiel

### CODE REVIEW

File: library/tabs/src/TabsWrapper.php [hunk 1]
```diff
@@ -68,6 +71,16 @@ public function genCss()
 }
 EOD;
         }
+
```
Comment (sunsetsystems) [inline library/tabs/src/TabsWrapper.php]: The following should be within the above "if (!defined('INCLUDED_TW_ONETIME_CSS'))" conditional block, so it is not repeated when there are multiple tab sets.

**Comment by sunsetsystems** [comment]
Thanks for the improvement. Just one change requested.

File: library/tabs/src/TabsWrapper.php (hunk 1 above)
Comment (matrix-amiel) [reply library/tabs/src/TabsWrapper.php]: Done.

File: library/tabs/src/TabsWrapper.php [hunk 2]
```diff
@@ -60,14 +60,27 @@ public function genCss()
...
             define(INCLUDED_TW_ONETIME_CSS, true);
             $s .= <<<EOD
-.ui-tabs .ui-tabs-panel {
-  padding: 0px 0px;
-}
-.ui-tabs .ui-tabs-nav .ui-tabs-anchor {
-  padding: .1em .4em;
-}
-EOD;
+        #enctabs-wrapper{
```
Comment (sunsetsystems) [inline library/tabs/src/TabsWrapper.php]: Actually the TabsWrapper class should not have anything that is specific to just one (enctabs) instance of it. I think this style rule belongs in encounter_top.php until a better place is found.

File: library/tabs/src/TabsWrapper.php [hunk 3]
```diff
@@ -60,14 +60,27 @@ public function genCss()
...
+        .ui-tabs .ui-tabs-nav .ui-tabs-anchor {
+          padding: .1em .4em;
+        }
+EOD;
+            if ($_SESSION['language_direction'] == 'rtl') {
+                $s .= <<<EOD
+        .ui-tabs { direction: rtl; }
+        .ui-tabs .ui-tabs-nav li.ui-tabs-tab {float: right; }
+        .ui-tabs .ui-tabs-nav li a { float: right; }
+        }
```
Comment (sunsetsystems) [inline library/tabs/src/TabsWrapper.php]: I don't see that this closing brace matches anything.

File: library/tabs/src/TabsWrapper.php [hunk 4]
```diff
@@ -60,14 +60,27 @@ public function genCss()
...
+EOD;
+            if ($_SESSION['language_direction'] == 'rtl') {
+                $s .= <<<EOD
+        .ui-tabs { direction: rtl; }
+        .ui-tabs .ui-tabs-nav li.ui-tabs-tab {float: right; }
+        .ui-tabs .ui-tabs-nav li a { float: right; }
+        }           
+EOD;
+            }
+}
```
Comment (sunsetsystems) [inline library/tabs/src/TabsWrapper.php]: Looks like this should be indented.

File: library/tabs/src/TabsWrapper.php [hunk 5]
```diff
@@ -60,14 +60,27 @@ public function genCss()
...
-  padding: 0px 0px;
-}
-.ui-tabs .ui-tabs-nav .ui-tabs-anchor {
-  padding: .1em .4em;
-}
-EOD;
+        #enctabs-wrapper{
+          margin: 0px;
         }
+        .ui-tabs .ui-tabs-panel {
```
Comment (sunsetsystems) [inline library/tabs/src/TabsWrapper.php]: Should PHP output be indented to reflect the way that the generating PHP code is indented? I would vote not. You don't need to change this, just a future topic for discussion.

File: library/tabs/src/TabsWrapper.php (hunk 2 above)
Comment (matrix-amiel) [reply library/tabs/src/TabsWrapper.php]: Done.

File: library/tabs/src/TabsWrapper.php (hunk 3 above)
Comment (matrix-amiel) [reply library/tabs/src/TabsWrapper.php]: Done.

**Comment by matrix-amiel** [comment]
Hi @sunsetsystems 
Hopefully it's ready to merge now. We need that as soon as possible.
Thank you.
Amiel

**Comment by sunsetsystems** [comment]
Looks good! Thanks for the contribution.

**Comment by sunsetsystems** [comment]
@bradymiller Looks like Travis did not run for this PR?

**Comment by bradymiller** [comment]
@sunsetsystems ,
Yep, travis isn't working for the matrix folks. We looked into ad travis let us know that all the matrix developers need to log into travis for it to work; I doubt this will help, but we'll see if that works though. For the time being, just keep bringing their stuff in as normal (I can quickly fix up PSR2 issues in the codebase).

**Pull request merged**